pkg/action/install.go

```diff
@@ -344,8 +344,10 @@ func (i *Install) RunWithContext(ctx context.Context, chrt *chart.Chart, vals ma
 		return rel, err
 	}
 	rChan := make(chan resultMessage)
+	doneChan := make(chan struct{})
+	defer close(doneChan)
 	go i.performInstall(rChan, rel, toBeAdopted, resources)
-	go i.handleContext(ctx, rChan, rel)
+	go i.handleContext(ctx, rChan, doneChan, rel)
 	result := <-rChan
```

Comment:
I have a question about the process sequence. Imagine a case in RunWithContex()
RunWithContex() {
go i.performInstall // running for a longer time
go i handleContex // return from ctx.Done() and send message to rchan, but performInstall is always running
result := < -rChan //read message
}

the message in rchan will be received in RunWithContex(), and RunWithContex() returns soon.
so, will this case lead to goroutine performInstall leak?